src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs

```diff
@@ -205,17 +208,21 @@ private ObjectTypeConstructor BuildInitializationConstructor()
                 }
                 else if (IsStruct || property.SchemaProperty?.IsRequired == true)
                 {
-                    // For structs all properties become required
-                    Constant? defaultParameterValue = null;
                     if (property.SchemaProperty?.ClientDefaultValue is object defaultValueObject)
                     {
                         defaultParameterValue = BuilderHelpers.ParseConstant(defaultValueObject, propertyType);
                     }
 
+                    var inputType = TypeFactory.GetInputType(propertyType);
+                    if (defaultParameterValue != null && TypeFactory.IsStruct(property.ValueType))
```

Comment:
Should we move the entire `CanBeInitializedInline` to the type factory?

Reply:
But is doesn't take `CSharpType` param. I thought  type factory is for all CSharpType related methods?

Reply:
We can throw the method on BuilderHelpers too. It just seems that every time you say `IsStruct` what you are  really want to know is `CanBeInitializedInline `

Reply:
We need `IsStruct` condition before creating an object of `Parameter` and `CanBeInitializedInline` takes `Parameter` as a param. So I think we should keep it as is.

Reply:
But is the condition here specific to enums only or would we need similar handling, for example, for `Uris` that are also handled by `CanBeInitializedInline`?

We can also change the method signature.

Reply:
This is specific to enums only.

Reply:
Yes, we need similar handling for `Uris` as well. Moved `CanBeInitializedInline` to the type factory.